tests/test_csr.py

```diff
@@ -241,3 +246,193 @@ def test_admin_save_existing_csr_should_not_renew(admin_client):
+        organization_name="Test Organization",
+        state_or_province_name="Test State",
+        country_name="NL",
+        email_address="email@valid.com",
+    )
+
+    # Use a different private key to generate the certificate with an invalid signature
+    private_key = generate_private_key()
+    private_key = load_pem_x509_private_key(private_key.encode())
+    # Here I did not use the fixture mkcert to generate the certificate
+    # This is because the fixture mkcert did not generate a valid certificate
+    # The public keys of the private key and the certificate did not match
+    pub_cert = (
+        x509.CertificateBuilder()
+        .subject_name(
+            x509.Name(
+                [x509.NameAttribute(x509.NameOID.COMMON_NAME, "test.example.com")]
+            )
+        )
+        .issuer_name(
+            x509.Name(
+                [x509.NameAttribute(x509.NameOID.COMMON_NAME, "test.example.com")]
+            )
+        )
+        .public_key(private_key.public_key())
+        .serial_number(x509.random_serial_number())
+        .not_valid_before(datetime.datetime.now())
+        .not_valid_after(datetime.datetime.now() + datetime.timedelta(days=365))
+        .add_extension(
+            x509.SubjectAlternativeName([x509.DNSName("localhost")]),
+            critical=False,
+        )
+        .sign(private_key, hashes.SHA256(), default_backend())
+    )
+
+    cert_bytes = pub_cert.public_bytes(serialization.Encoding.PEM)
+    cert_pem = SimpleUploadedFile("cert.pem", cert_bytes)
+
+    form_data = {
+        "common_name": "test.example.com",
+        "organization_name": "Test Organization",
+        "country_name": "NL",
+        "state_or_province_name": "Test State",
+        "email_address": "email@valid.com",
+        "certificate": cert_pem,
+    }
+
+    # Valid certificate should create a Certificate instance
+    assert pub_cert.public_key() == private_key.public_key()
+
+    # Saving the same certificate again should not create a new instance
+    response = admin_client.post(
+        f"/admin/simple_certmanager/signingrequest/{csr.pk}/change/",
+        data=form_data,
+    )
+    assert response.status_code == 200
+    assert len(response.context["adminform"].form.errors) > 0
+    assert (
+        "Certificate does not match the signature from the actual CSR."
+        in response.context["adminform"].form.errors["certificate"]
+    )
+    assert Certificate.objects.count() == 0
+
+
+@pytest.mark.django_db
+def test_saving_invalid_cert_does_not_create_cert_instance_via_post(
+    admin_client,
+    temp_private_root,
+):
+    assert Certificate.objects.count() == 0
+
+    csr = SigningRequest.objects.create(
+        common_name="test.example.com",
+        organization_name="Test Organization",
+        state_or_province_name="Test State",
+        country_name="NL",
+        email_address="email@valid.com",
+    )
+
+    cert_pem = ContentFile(b"invalid bytes", "cert.pem")
```

Comment:
could also be a `SimpleUploadedFile`, it's a bit odd that different file classes are used now in tests

```suggestion
    cert_pem = SimpleUploadedFile("cert.pem", b"invalid bytes")
```